ext/standard/syslog.c

```diff
@@ -152,6 +152,7 @@ PHP_FUNCTION(openlog)
 		RETURN_FALSE;
 	}
 	openlog(BG(syslog_device), option, facility);
+	PG(have_called_openlog) = 1;
```

Comment:
I would suggest one small improvement. Maybe we could introduce `php_openlog` with the same args as openlog that would just call openlog and set `PG(have_called_openlog)` so it's done internally in that function. Then it would be used in here and fpm.

Reply:
Then... `syslog()` would need to call `php_openlog()`...  What would that change?

Reply:
Then... syslog() would need to call php_openlog()... What would that change?

Reply:
What I'm saying is to just replace:
```
openlog(BG(syslog_device), option, facility);
PG(have_called_openlog) = 1;
```
with
```
php_openlog(BG(syslog_device), option, facility);
```
and implement php_openlog as 
```
void php_openlog(const char *ident, int option, int facility)
{
    openlog(BG(syslog_device), option, facility);
    PG(have_called_openlog) = 1;
}
```
Then you also replace it in other place. It's just a wrapper as you use the same pattern in 2 other places - no change in your logic...